### PR TITLE
Implement freshness checks using the Echo option

### PR DESCRIPTION
**Pull request by tiniuclx**

### CODE REVIEW

**Comment by tiniuclx** [comment]
~format

**Comment by tiniuclx** [comment]
Just tested the KNX-IoT Virtual demo using an echo-enabled build of KNX, and the latest source code from knx-iot-virtual and knx-iot-point-api-config. The virtual push button and actuator can be programmed as expected. Once they are both loaded, they can still communicate to each other, though the first few transactions take a little bit longer as the freshness of the request is checked using the Echo option, per the KNX spec. However, the first button press is still successful - it correctly toggles the actuator on the other side.

File: messaging/coap/coap.h [hunk 1]
```diff
@@ -104,7 +104,7 @@ typedef struct
   uint8_t token_len;
   uint8_t token[COAP_TOKEN_LEN];
 
-  uint8_t options[COAP_OPTION_SIZE1 / OPTION_MAP_SIZE +
+  uint8_t options[COAP_OPTION_ECHO / OPTION_MAP_SIZE +
                   1]; /* bitmap to check if option is set */
```
Comment (WAvdBeek) [inline messaging/coap/coap.h:108]: we need to document this file with doxygen

File: messaging/coap/constants.h [hunk 2]
```diff
@@ -78,6 +78,7 @@ extern "C" {
   4 /* | version:0x03 type:0x0C tkl:0xF0 | code | mid:0x00FF | mid:0xFF00 | */
 #define COAP_TOKEN_LEN 8 /* The maximum number of bytes for the Token */
```
Comment (WAvdBeek) [inline messaging/coap/constants.h:79]: same here e.g. uses /**< */

**Review by WAvdBeek — APPROVED** [review]

**Pull request merged**